libhdf5/hdf5attr.c

```diff
@@ -514,8 +514,16 @@ NC4_put_att(int ncid, int varid, const char *name, nc_type file_type,
          else
             *(char **)var->fill_value = NULL;
       }
-      else
-         memcpy(var->fill_value, data, type_size);
+      else {
+         if (var->type_info->nc_type_class == NC_OPAQUE ||
+             var->type_info->nc_type_class == NC_COMPOUND ||
+             var->type_info->nc_type_class == NC_ENUM)
+             memcpy(var->fill_value, data, type_size);
+         else if ((retval = nc4_convert_type(data, var->fill_value, mem_type,
+                                        file_type, len, &range_error, NULL,
+                                        (h5->cmode & NC_CLASSIC_MODEL))))
+             BAIL(retval);
+      }
```

Comment:
OK, I'm sorry that this PR did not manage to get merged, because there seems to be a lot of important things going on here. I would like to get some of this work on one of my branches so that it does not get lost.

@wkliao can you explain what is going on with these new lines to NC4_put_att? Are they tested anywhere in this PR?